src/tracer.js

```diff
@@ -0,0 +1,181 @@
+'use strict';
+
+import * as Constants from './constants';
+import Span from './span';
+
+/**
+ *
+ */
+export default class Tracer {
+
+    /**
+     * [constructor description]
+     * @return {[type]} [description]
+     */
+    constructor() {
+        this._imp = null;
+    }
+
+    /**
+     * Handle to implementation object.
+     */
+    imp() {
+        return this._imp;
+    }
+
+    /**
+     * TODO: Unclear what the OpenTracing specification wants here.  This
+     * implementation is a best-guess approximation that's likely not right.
+     * The spec refers to text and binary representations, but what does
+     * "binary" mean in JavaScript without making assumptions about protocols,
+     * transports, and platforms?
```

Comment:
As the code comment here indicates, I was scratching my head a bit as to what the intention of the TraceContext encode / decode *specifically* to text and binary formats was for. Put differently, I wholeheartedly understand and support the need for the ability to serialize a TraceContext, but didn't quite understand the binary case (in particular) well enough to know how to implement this API in a general fashion in JS.

That's likely best discussed as part of the spec documentation rather than here, but wanted to note it here as this code undoubtedly needs to be revised once I understand the intention a bit better!